src/iter/mod.rs

```diff
@@ -843,7 +852,47 @@ pub trait ParallelIterator: Sized + Send {
         FlatMap::new(self, map_op)
     }
 
-    /// An adaptor that flattens iterable `Item`s into one large iterator
+    /// Applies `map_op` to each item of this iterator to get nested serial iterators,
+    /// producing a new parallel iterator that flattens these back into one.
+    ///
+    /// Each nested iterator will be flattened local to the thread where the mapping takes
```

Comment:
Maybe add a section like

### `flat_map` vs `flat_map_iter`

These two methods are similar but behave slightly differently. With `flat_map`, each of the nested iterators must be a parallel iterator, and ultimately we will parallelize all the nested items independently. With `flat_map_iter`, each nested iterator can be a sequential iterator, and we only parallelize *between iterators* (but all items produced by any iterator are processed sequentially).

If the iterators that are being produced tend to have relatively few items, or a consistent number of items, `flat_map_iter` will likely perform better, as it has less parallelization overhead. If you really need to be able to parallelize between the individual items (and not just between iterators), then you should choose `flat_map`.